### DISCUSSION TITLE
Turn off recombination

### DISCUSSION THREAD

**Discussion by Anilkumargpb** [gaynorr/AlphaSimR#194] [General]
Hi Gaynor,

Is it possible to turn off recombination while simulating the population? If possible, please share the function. 

Thanks 
Anil

**Comment by gaynorr** [comment]
Hi @Anilkumargpb,

Sorry for not responding earlier.

You can adjust the genetic map at any time in the simulation. Thus, you can effectively turn off recombination by setting all the genetic map positions to 0. The software will still go through the steps of modeling recombination, but there would never be an opportunity for recombination to occur. Note that sampling of chromosomes during meiosis will still occur. Below is some example code.

```
library(AlphaSimR)

founderPop = quickHaplo(nInd=2, nChr=1, segSites=10, 
                        inbred=TRUE)

# Turning recombination tracking on to be able to use
# pullIbdHaplo
SP = SimParam$
  new(founderPop)$
  setTrackRec(TRUE) 


# Setting all map positions to zero and saving original map
origMap = SP$genMap
zeroMap = lapply(origMap, function(x) x*0)
SP$switchGenMap(zeroMap)

# Create initial parents and make F2s
Parents = newPop(founderPop)
F1 = randCross(Parents, nCrosses=1)
F2 = self(F1, nProgeny=30)

# Examine haplotypes, there will be no recombination
# There should only be rows of 1s and 2s
pullIbdHaplo(Parents)
pullIbdHaplo(F1)
pullIbdHaplo(F2)

# Return to original map and recreate F2s
SP$switchGenMap(origMap)
F2r = self(F1, nProgeny=30)

# You should now see recombinations
pullIbdHaplo(F2r)

```

**Comment by Anilkumargpb** [comment]
Hello Chris,

Thanks for your response and information on blocking recombinations while
simulating.

Thanks

*डॉ. अनिल कुमार सी Dr. Anilkumar C.*

वैज्ञानिक (आनुवंशिकी एवं पौध प्रजनन)

Scientist (Genetics & Plant Breeding)

फसल उन्नयन प्रभाग Crop Improvement Division

भाकृअनुप-राष्ट्रीय चावल अनुसंधान संस्थान, कटक, भारत -753006

ICAR-National Rice Research Institute, Cuttack, Odisha, India -753006


*Fulbright Postdoc Fellow*

Dept. of Agronomy and Plant Genetics

University of Minnesota, St. Paul, MN, USA

<https://www.linkedin.com/in/anilkumar-c-85a4a497/>
